Issues/13　コメント投稿のUI作成

keitan1130 opened the pull request.

keitan1130 committed:
feat: update comment type and mock data

keitan1130 committed:
feat: implement reply function

keitan1130 committed:
style: remove unused font import

keitan1130 committed:
style: update pagination button color

Semi-koron commented on src/features/Pagenation/Button/index.module.css:
```diff
@@ -34,7 +34,7 @@
 }
 
 .page-btn--active {
-  background: #f6b000; /* 黄色 */
```
この黄色はできればパレットに登録しているプライマリーカラーを使って反映して欲しい。
https://github.com/simesaba80/toybox-front/blob/47dd1944a5039db499f664d7c2b7f4753428e52b/src/index.css#L7


```suggestion
  background: var(--primary-color);
```

Semi-koron commented on src/features/Pagenation/PageDropdown/index.module.css:
```diff
@@ -102,10 +102,10 @@
 }
 
 .itemActive {
-  background: #f6b000;
```
こちらもパレットカラーから設定お願いします

Semi-koron replied:
これ一度approveした後に申し訳ないのだけれど、クラス名はケバブケースに統一して欲しいです。item-activeのように-で単語を繋ぐようにお願い🙏

Semi-koron commented on src/features/Pagenation/PageDropdown/index.module.css:
```diff
@@ -102,10 +102,10 @@
 }
 
 .itemActive:hover {
-  background: #f6b000;
```
こちらもパレットカラーから設定お願いします

Semi-koron commented on src/pages/WorkPage/CommentSection/CommentInput/SendButton/index.module.css:
```diff
@@ -0,0 +1,26 @@
+.sendButton {
+  width: 106px;
+  height: 54px;
+  background: #fabe00;
```
こちらもパレットカラーから設定お願いします

Semi-koron commented on src/pages/WorkPage/CommentSection/CommentInput/SendButton/index.tsx:
```diff
@@ -0,0 +1,23 @@
+import type React from "react";
+
+interface SendButtonProps {
+  onClick?: () => void;
```
オプションである必要はないのでこちらは必須項目に


```suggestion
  onClick: () => void;
```

Semi-koron commented on src/pages/WorkPage/CommentSection/CommentInput/SendButton/index.tsx:
```diff
@@ -0,0 +1,23 @@
+  );
+};
+
+export default SendButton;
```
SendButtonとしてコンポーネントを区切るのではなく、Buttonコンポーネントとして区切るべきなので、
送信という文言を変えれるようにPropsにReactNodeを持たせれるようにすると今後ボタン作るぞってなった時に再利用しやすい。
一方でもうすでにページネーションで[Buttonが宣言されている](https://github.com/simesaba80/toybox-front/blob/47dd1944a5039db499f664d7c2b7f4753428e52b/src/features/Pagenation/Button/index.tsx#L1-L45)が、こちらはButtonという名前を使うには用途が限定されているので、こちらのボタンの名前をPagenationButtonなどに変更して、名前衝突の回避もお願いしたい。

Semi-koron commented on src/pages/WorkPage/CommentSection/CommentInput/index.module.css:
```diff
@@ -0,0 +1,106 @@
+.inputRow {
```
こちらのファイルもケバブケースに変更お願いします

Semi-koron commented on src/pages/WorkPage/CommentSection/CommentInput/index.tsx:
```diff
@@ -0,0 +1,100 @@
+
+import type React from "react";
+import type { Comment } from "../../../../shared/types/comment";
+
+interface CommentInputProps {
+  onSubmit?: (message: string) => void;
+  replyingTo?: Comment | null;
+  onCancelReply?: () => void;
+  autoFocus?: boolean;
+}
```
onSubmitはオプショナルじゃなくて良いかな
あとavatarUrlとかをオプショナルで持たせておくと良いかも
```suggestion
interface CommentInputProps {
  onSubmit: (message: string) => void;
  replyingTo?: Comment | null;
  onCancelReply?: () => void;
  autoFocus?: boolean;
  avatarUrl?: string;
}
```

Semi-koron commented on src/pages/WorkPage/CommentSection/CommentList/CommentItem/index.module.css:
```diff
@@ -0,0 +1,137 @@
+  padding: 18px 20px;
+}
+
+.bubbleText {
```
こちらもケバブケースお願いします

Semi-koron commented on src/pages/WorkPage/CommentSection/CommentList/CommentItem/index.tsx:
```diff
@@ -0,0 +1,132 @@
+interface CommentItemProps {
+  comment: Comment;
+  onDelete?: (commentId: string) => void;
+  onReply?: (comment: Comment) => void;
```
onReplyは必ずできると思うので必須項目に変更お願いします

Semi-koron commented on src/pages/WorkPage/CommentSection/CommentList/CommentItem/index.tsx:
```diff
@@ -0,0 +1,132 @@
+  replies?: Comment[];
+  allComments?: Comment[];
+  depth?: number;
+  replyingTo?: Comment | null;
```
nullがあるならオプショナルである必要がないのかなあと思っているので、nullを外すか、オプショナルを外すかのどっちかがいい。

Semi-koron commented on src/features/CommentSection/CommentList/CommentItem/index.tsx:
```diff
@@ -0,0 +1,132 @@
+import type React from "react";
+import type { Comment } from "../../../../../shared/types/comment";
+
+interface CommentItemProps {
+  comment: Comment;
+  onDelete?: (commentId: string) => void;
+  onReply?: (comment: Comment) => void;
+  replies?: Comment[];
+  allComments?: Comment[];
+  depth?: number;
+  replyingTo?: Comment | null;
+  onSubmitReply?: (message: string, parentId?: string) => void;
+  onCancelReply?: () => void;
+}
```
関数のバケツリレーが起こっているから今回は直さなくて良いんだけど、zustandを使ってバケツリレーの解消したい

Semi-koron reviewed (commented):
pages配下のファイルでは基本的にfeaturesディレクトリでできたコンポーネントを組み立てるだけにしたいので、CommentSectionはfeaturesに移動お願いします。

keitan1130 committed:
refactor: move CommentSection to features directory

keitan1130 committed:
refactor: replace custom buttons with shared Button component and use path aliases

keitan1130 committed:
style: apply color palette from CSS custom properties

keitan1130 committed:
refactor: convert CSS class names to kebab-case in CommentSection and Pagenation

keitan1130 committed:
refactor: improve type safety in CommentSection components

- Make onClick required in Button component
- Make onSubmit required in CommentInput component
- Make onReply required in CommentItem and CommentList components
- Replace null with undefined for optional replyingTo state
- Add avatarUrl prop to CommentInput for customizable avatar
- Remove optional chaining for required callback props

keitan1130 committed:
refactor: remove Button component and rename btn to button in class names

keitan1130 committed:
refactor: rename btn to button in CSS class names

Semi-koron reviewed (commented):
LGTM
